internal/server/server_test.go

```diff
@@ -252,6 +252,11 @@ func TestServer_GenerateRoutes_NoRoute(t *testing.T) {
 
 	run := func(t *testing.T, tc testCase) {
 		req := httptest.NewRequest(http.MethodGet, tc.path, nil)
+
+		if tc.name != "no header" {
+			req.Header.Set("Accept", tc.name)
+		}
```

Comment:
I think it might be nice to keep the accept header separate from the name.  I suspect it won't be too obvious that the name is always used as the accept header when some test names don't look like the value for that header.

One pattern we've used in other places is to add another field to the `testCase` struct.

```go
setup func(t *testing.T, req *http.Request)
```

And here we'd do something like this:

```suggestion
		if tc.setup != nil {
			tc.setup(t, req)
		}
```